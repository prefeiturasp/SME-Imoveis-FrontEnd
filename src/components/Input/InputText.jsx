import PropTypes from "prop-types";
import React from "react";
import { InputErroMensagem } from "./InputErroMensagem";
import { HelpText } from "components/HelpText";
import { OverlayTrigger, Popover, Image } from "react-bootstrap";
import "./style.scss";
import imgTooltipQuestion from "img/tooltip-question.svg";
export class InputText extends React.Component {
  popover = msg => (
    <Popover id="popover-basic">
      <Popover.Content>{msg}</Popover.Content>
    </Popover>
  );

  render() {
    const {
      className,
      disabled,
      esconderAsterisco,
      helpText,
      input,
      label,
      labelClassName,
      meta,
      min,
      name,
      placeholder,
      required,
      type,
      maxLength,
      customChange,
      autoFocus,
      tooltipMessage
    } = this.props;
    return (
      <div className="input">
        {label && [
          required && !esconderAsterisco && (
            <span key={1} className="required-asterisk">
              *
            </span>
          ),
          <label
            key={2}
            htmlFor={name}
            className={`col-form-label ${labelClassName}`}
          >
            {label}
          </label>,
          tooltipMessage && (
            <OverlayTrigger
              trigger="click"
              placement="right"
              overlay={this.popover(tooltipMessage)}
            >
              <Image src={imgTooltipQuestion} fluid className="mb-1 ml-1" />
            </OverlayTrigger>
          )
        ]}
        <input
          {...input}
          autoFocus={autoFocus}
          className={`form-control ${className} ${meta &&
            meta.touched &&
            (meta.error || meta.warning) &&
            "invalid-field"}`}
          disabled={disabled}
          min={min}
          name={name}
          placeholder={placeholder}
          type={type}
          maxLength={maxLength}
          {...(() => {
            if (customChange) {
              return {
                onChange: e => {
                  input.onChange(customChange(e.target.value));
                }
              };
            } else {
              return {};
            }
          })()}
        />
        <HelpText helpText={helpText} />
        <InputErroMensagem meta={meta} />
      </div>
    );
  }
}

InputText.propTypes = {
  className: PropTypes.string,
  disabled: PropTypes.bool,
  esconderAsterisco: PropTypes.bool,
  helpText: PropTypes.string,
  input: PropTypes.object,
  label: PropTypes.string,
  labelClassName: PropTypes.string,
  meta: PropTypes.object,
  name: PropTypes.string,
  placeholder: PropTypes.string,
  required: PropTypes.bool,
  type: PropTypes.string
};

InputText.defaultProps = {
  className: "",
  disabled: false,
  esconderAsterisco: false,
  helpText: "",
  input: {},
  label: "",
  labelClassName: "",
  meta: {},
  name: "",
  placeholder: "",
  required: false,
  type: "text"
};
