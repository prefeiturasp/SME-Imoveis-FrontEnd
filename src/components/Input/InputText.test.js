import { mount } from "enzyme";
import React from "react";
import { InputText } from "./InputText";

describe("test <InputText>", () => {
  let wrapper;
  beforeAll(() => {
    wrapper = mount(<InputText />);
  });

  it("renders component", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
