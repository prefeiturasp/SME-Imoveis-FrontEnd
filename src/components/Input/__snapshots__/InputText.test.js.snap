// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test <InputText> renders component 1`] = `
<InputText
  className=""
  disabled={false}
  esconderAsterisco={false}
  helpText=""
  input={Object {}}
  label=""
  labelClassName=""
  meta={Object {}}
  name=""
  placeholder=""
  required={false}
  type="text"
>
  <div
    className="input"
  >
    <input
      className="form-control  undefined"
      disabled={false}
      name=""
      placeholder=""
      required={false}
      type="text"
    />
    <HelpText
      helpText=""
    >
      <div
        className="help-text"
      />
    </HelpText>
    <InputErroMensagem
      meta={Object {}}
    >
      <div
        className="error-or-warning-message"
      />
    </InputErroMensagem>
  </div>
</InputText>
`;
